### PR TITLE
Bgs liveness

### PR DESCRIPTION
**Pull request by ponny-m**
## What was the problem?
Implementing K8s Liveness and Readiness on BGS-api (Ruby application)

Associated tickets or Slack threads:
- #2417 

## How does this fix it?[^1]
This configures K8s liveness and readiness for svc-bgs-api to ensure highest level of visibility and reliability into the health and readiness of platform and partner team services.

## How to test this PR
- Step 1
    - Use Lens to check Pod Descriptions for Readiness Probe Output
    
- Step 2
    - Simulate Probe Failure to test if the probes are working correctly. 
    - **For Liveness:** Disrupt connection between RabbitMQ and BGS-api 
    - **For Readiness:** Temporarily disrupt the connection to RabbitMQ and also disrupt BGS service and make it unavailable 
       or unresponsive. 
    - Check if BOTH liveness and readiness probes correctly identifies these. Pods will restart if Liveness probe fails while 
       pods will be marked as 'Not Ready' if Readiness probe fails.

### CODE REVIEW

**Comment by ponny-m** [comment]
When reviewing focus on the last two commits. I also reviewed the logs for the failed tests and they aren't related to any changes made in the PR so I think this is good to go.

**Review by msnwatson — APPROVED** [review]

File: svc-bgs-api/src/healthcheck/liveness_script.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+def check_rabbitmq_connection
+  subscriber = RabbitSubscriber.new(BUNNY_ARGS)
+  subscriber.connected?
+rescue StandardError => e
+  puts "Liveness check failed: #{e.message}"
+  false
+end
+
+exit check_rabbitmq_connection ? 0 : 1
```
Comment (tejans24) [inline svc-bgs-api/src/healthcheck/liveness_script.rb]: I think it helps to have consistency on exists, this vs in svc-bgs-api/src/healthcheck/readiness_script.rb
```
if rabbitmq_connection_active? && bgs_service_available?
  exit 0
else
  puts "Readiness check failed"
  exit 1
end
```

**Review by tejans24 — APPROVED** [review]
LGTM added a comment to correct

**Review by ponny-m — COMMENTED** [review]
Made adjustments to ensure consistency

File: svc-bgs-api/src/healthcheck/liveness_script.rb (hunk 1 above)
Comment (ponny-m) [reply svc-bgs-api/src/healthcheck/liveness_script.rb]: Made adjustments to ensure consistency

**Pull request merged**